### ISSUE TITLE
Theme Change does not work for collections 

### ISSUE THREAD

**Issue by KWodarczyk** [microsoft/microsoft-ui-xaml#8433]
[Uploading WinUI3TestApp2.zip…]()
### Describe the bug

Theme Change does not work for collections. when theme is changed the values from the theme dictionaries are not used but only if the control is inside collection. It works fine if used a stand alone .

```

    <StackPanel Orientation="Horizontal" HorizontalAlignment="Center" VerticalAlignment="Center">
        <Button Content="Dark Theme" x:Name="darkbtn" Background="Black" Foreground="White" Click="darkbtn_Click" Margin="10"/>
        <Button Content="Light Theme" x:Name="lightbtn" Background="WhiteSmoke" Foreground="DarkGray" Click="lightbtn_Click" Margin="10"/>


        <local:UserControl1>
            
            <!--same contorl in a list not working-->
            <local:UserControl1.NestedControls>
                <local:TestControl SomeBursh="{ThemeResource SomeBurhs}" MyProperty="{ThemeResource SomeString}"/>
            </local:UserControl1.NestedControls>
            
            <!--same control as singel property this works fine-->
            <local:UserControl1.NestedControl>
                <local:TestControl SomeBursh="{ThemeResource SomeBurhs}" MyProperty="{ThemeResource SomeString}"/>
            </local:UserControl1.NestedControl>
        </local:UserControl1>
    </StackPanel>
```

### Steps to reproduce the bug

just run the project attached 

### Expected behavior

colors and string should change everywhere 

### Screenshots

![image](https://user-images.githubusercontent.com/5243438/234663201-71164b40-ff14-4e29-9435-a644e46a01d7.png)

![image](https://user-images.githubusercontent.com/5243438/234663703-76a1b3f6-d771-4338-9754-bf7c49075a90.png)

### NuGet package version

WinUI 3 - Windows App SDK 1.3.0: 1.3.230331000

### Windows version

Windows 11 (21H2): Build 22000

**Comment by KWodarczyk** [comment]
[WinUI3TestApp2.zip](https://github.com/microsoft/microsoft-ui-xaml/files/11336032/WinUI3TestApp2.zip)

**Comment by JJBrychell** [comment]
It is hard to understand what you are attempting to accomplish here, but things seem to be working as expected.

The issue that appears to be happening is that items that are not part of the visual tree are not subject to theme resource updates. When a theme change occurs, the visual tree is walked and all the properties on each element are checked to see their values need to be updated.  If an item is not in the visual tree, it will never be hit and its values never updated.

Now, the visual tree consists of all elements that directly contribute to the visual representation of the scene.  It does not consist of any databound objects whose properties may or may not be used to construct the tree and in this case and as we can see from this snippet of the visual tree from the sample app, TestControl does not appear in the visual tree:
<code>
CUserControl, class WinUI3TestApp2.UserControl1
&nbsp;&nbsp;CBorder
&nbsp;&nbsp;&nbsp;&nbsp;CStackPanel
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CButton
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CContentPresenter "ContentPresenter"
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CTextBlock, text {light str}
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CTextBlock "textBlockTest", text {light str}
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CGrid
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CCanvas "testCanva"
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CRectangle
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CRectangle
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;…
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CRectangle
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CRectangle
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CListView
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CBorder
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CScrollViewer "ScrollViewer"
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CBorder "Root"
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CGrid
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CScrollContentPresenter "ScrollContentPresenter"
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CItemsPresenter
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CContentControl
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CItemsStackPanel
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CListViewItem
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CListViewItemChrome "Root", class Microsoft.UI.Xaml.Controls.Primitives.ListViewItemPresenter
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CBorder
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CStackPanel
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CButton
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CContentPresenter "ContentPresenter"
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CTextBlock, text {light str}
</code>
But instead. it is used a data source to populate the ListView.  And since TestControl does not exist in the visual tree, its property values are evaluated only once when it is defined in the markup:
<code>
&lt;local:UserControl1.NestedControls>
&nbsp;&nbsp;&nbsp;&nbsp;&lt;local:TestControl SomeBursh="{ThemeResource SomeBurhsResource}" MyProperty="{ThemeResource SomeString}"/>
&lt;/local:UserControl1.NestedControls>
</code>
It doesn't matter that TestControl is derived from a Xaml control, because at this point, it is just seen as a dependency object in a collection of dependency objects being use as a data source and treated no different than a simple data class in a collection of simple data classes.  

I am leery of offering any alternate solutions as this is a rather unique use of xaml (custom collection of controls as a data source to a listview) and so I am not sure of the actual objective you are trying to achieve.

**Comment by JJBrychell** [comment]
We are going to close this issue as by design.  If you end up with a more understandable repro we can help address alternatives in the future.

**Comment by DmitriyKomin** [comment]
@KWodarczyk, resolving per @JJBrychell's comments.
